docs/docsite/rst/playbook_guide/playbooks_loops.rst

```diff
@@ -256,6 +256,41 @@ When you run a task with ``until`` and register the result as a variable, the re
 
 If ``until`` is not specified, the task will retry until the task succeeds but at most ``retries`` times.
 
+.. _retry_with_timeout:
+
+Retrying a task with timeout
+----------------------------
+
+You can use the ``timeout`` keyword to set a time limit for the execution of the task code. 
+
+When you retry a task, ``timeout`` applies per each attempt and remains consistent -- it is not affected by the total time spent on a task.
+
+For example
+
+.. code-block:: yaml
+
+    - name: Retry a task with timeout
+      ansible.builtin.shell: sleep 10; /usr/bin/foo
+      register: result
+      until: result != -1
+      retries: 3
+      timeout: 15
```

Comment:
I feel like there could be another example that illustrates the behaviour better than the one from the issue. How about something like this?

```
- name: Ensure the service is running
  ansible.builtin.service:
    name: my_service
    state: started
  register: service_status
  until: service_status is succeeded
  retries: 5
  delay: 10
  timeout: 60
```

Here we use the same retries and delay as the preceding example. It's also maybe more "real world" than the one from the issue and easier to understand. This tries to start "my_service" and retries the operation up to five times with a delay of 10 seconds between each retry attempt. Each attempt has 60 seconds to complete.